GalilSup/Db/galil_motor_extras.template

```diff
@@ -1018,24 +1018,27 @@ record(stringout,"$(P):$(M)_MOVE_CMD")
 
 ##
 ## calculate the live difference between motor and encoder position
-## only calculate if UEIP is true and an encoder is actually
+## only calculate if UEIP is true and an encoder is also actually
 ## present (as per relevant MSTA bit)
+## In addition only calculate when moving to avoid noisy
+## encoder post move triggering recording 
 ##
 record(calcout,"$(P):$(M)_MTRENC_DIFF")
 {
-	field(DESC, "Encoder motor difference")
+	field(DESC, "In motion encoder motor difference")
 	field(INPA, "$(P):$(M).RMP CP")
 	field(INPB, "$(P):$(M).MRES CP")
 	field(INPC, "$(P):$(M).REP CP")
 	field(INPD, "$(P):$(M).ERES CP")
 	field(INPE, "$(P):$(M).UEIP CP")
 	field(INPF, "$(P):$(M).MSTA CP")
-	field(CALC, "(E&&(F&256))?(ABS(A*B)-ABS(C*D)):0")
+	field(INPG, "$(P):$(M).DMOV CP")
+	field(CALC, "H:=(G=0?((E&&(F&256))?(ABS(A*B)-ABS(C*D)):0):H);H")
```

Comment:
This logic is non-trivial. Could we add an ioc test for it? Also can we add a code comment breaking down what this is intended to do, as it's a bit difficult to read.

Reply:
I suspect a simulated motor might never show a difference between encoder and motor position, maybe changing MRES might force it to do something? Or would it be sufficient to write directly to the calc fields in the test?

Reply:
If it's difficult to simulate the inputs properly with the motor in sim mode then yes I think directly writing to the inputs for the test is fine. I am mostly concerned about testing the complex-looking calc expression rather than the plumbing which links it to other records.

Reply:
I've added a test, it required extending the motor record simulation framework as simulated motors did not have encoders